pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -208,11 +212,25 @@ func (a *Bootstrap) getTemplateData(installConfig *types.InstallConfig, proxy *c
 		logrus.Debugf("Using internal constant for release image %s", releaseImage)
 	}
 
+	var deployImage string
+	// baremetal requires some additional templateData to download the RHCOS image
+	// so we can deploy the masters with the correct image from rhcos.json
+	if installConfig.Platform.Name() == bmtypes.Name {
```

Comment:
No need for this to be bare metal specific - just make baseURI available for templating by all platforms

Reply:
Thanks for the feedback @markmc this is all now working locally for me so I kicked off a CI run via https://github.com/openshift-metal3/dev-scripts/pull/642 to replicate that, but in the meantime I will rework this to address your comments, and when you're happy we can break out that commit and propose upstream (unless @stbenjam plans to push a PR with this combination of commits soon, I don't want to complicate that..)